### PR TITLE
s3-credentials: init at 0.10

### PR DESCRIPTION
**Pull request by techknowlogick**
###### Description of changes

Add https://github.com/simonw/s3-credentials

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/s3-credentials/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    sha256 = "sha256-ieFprsDkxb2w6Qc/DQ58LIfZgZSu/i8ecM1ZnS9hUDs=";
+  };
+
+  propagatedBuildInputs = [
+    click
+    click-default-group
+    boto3
+  ];
+
+  # tests aren't included in the PyPI packages
```
Comment (fabaff) [inline pkgs/development/python-modules/s3-credentials/default.nix:26]: Upstream is [tagging](https://github.com/simonw/s3-credentials/releases) the source. Could you check if the tests could be executed by using `fetchFromGitHub` to get the release tarball?

Comment (techknowlogick) [reply pkgs/development/python-modules/s3-credentials/default.nix]: sounds good. the tests are now included and run :)

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 165909` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.s3-credentials</li>
    <li>s3-credentials (python39Packages.s3-credentials)</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/165909#issuecomment-1080271081)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-165909]$ ./results/python310Packages.s3-credentials/bin/s3-credentials --version
s3-credentials, version 0.10
```

Looks good to me.

**Pull request merged**